src/Data/Maybe.purs

```diff
@@ -141,6 +141,23 @@ instance bindMaybe :: Bind Maybe where
 -- | ``` purescript
 -- | x >>= (\x' -> y >>= (\y' -> pure (f x' y')))
 -- | ```
+-- |
+-- | and is conceptually similar to writing...
+-- | ```
+-- | var x = // ...
+-- | if x.isNothing {
+-- |   return x; // Nothing
+-- | } else {
+-- |   var x' = x.unwrapJust;
+-- |   var y = // ...
```

Comment:
Since `x` and `y` are free variables in the first expression, I think they should probably be free variables in the pseudocode too. My main concern, though, would be that `.isNothing`, `.unwrapJust`, and `wrapIntoJust` might not be clear enough: I'd guess the people who are able to work out what these functions/methods mean in this pseudocode are likely to be those who already understand what's going on anyway and don't need any additional explanation. I feel like it might be best to use prose rather than pseudocode here?

Reply:
(I get that my editing for core library docs can be quite demanding, so I'm happy to take a stab at this myself if you'd prefer.)

Reply:
I'd prefer not using prose. Some examples, like this one, I think are easier to understand when you can see two code examples next to one another. 

If you want to submit something, go for it!

I could also break down the `bind` statements to this before showing the JavaScript version:
```purescript
case x of
  Nothing -> Nothing
  Just x' -> case y of
    Nothing -> Nothing
    Just y' -> Just (f x' y')
```

Reply:
That’s a really good point. Inlining everything so that it’s just pattern matching sounds great to me!

Reply:
Ok, I'll update it.

Reply:
Updated. Just waiting for CI to pass.

Reply:
Ah, I was imagining that the pattern matching version would be instead of the pseudocode. I have to admit I'm sort of struggling to see the value of the pseudocode example in addition to the pattern matching version.

Reply:
Hm.... It has value for someone who is familiar with JavaScript or other C-family languages, who is still not yet comfortable with PureScript syntax.
Otherwise, I'm not sure it's needed anymore if they understand pattern matching.

Reply:
I removed it in latest commit, so we can get this merged. We can always add it again later if people still have trouble with what's currently written.